src/AutoRest.CSharp/Common/Input/InputTypes/InputModelType.cs

```diff
@@ -1,14 +1,13 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using System;
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.Runtime.CompilerServices;
 
 namespace AutoRest.CSharp.Common.Input
 {
-    internal record InputModelType(string Name, string? Namespace, string? Accessibility, string? Deprecated, string? Description, InputModelTypeUsage Usage, IReadOnlyList<InputModelProperty> Properties, InputModelType? BaseModel, IReadOnlyList<InputModelType> DerivedModels, string? DiscriminatorValue, string? DiscriminatorPropertyName, bool IsNullable)
+    internal record InputModelType(string Name, string? Namespace, string? Accessibility, string? Deprecated, string? Description, InputModelTypeUsage Usage, IReadOnlyList<InputModelProperty> Properties, InputModelType? BaseModel, IReadOnlyList<InputModelType> DerivedModels, string? DiscriminatorValue, string? DiscriminatorPropertyName, InputDictionaryType? InheritedDictionaryType, bool IsNullable)
```

Comment:
this change is taken from [Alexander's PR](https://github.com/Azure/autorest.csharp/pull/3726).

In most cases, it should work perfectly fine that we could get `InheritedDictionaryType` from the `BaseModel` property, especially in the scenario of using typespec, because in typespec, a model could only have one base model.
This is a backward compat thing because in swagger, a model could have multiple base models, and this property is not null when any of thme is a dictionary

Reply:
We could consider change this to a calculated property, for instance, for the case we have now, it could simply be:
```csharp
internal record InputModelType(/*parameters unchanged*/)
{
public InputDictionaryType InheritDictionaryType => BaseModel as InputDictionaryType;
}
```
and update the implementation here when we support the full conversion from swagger to input types.

But as for now, to reduce the amount of changes I will just leave it here, and may go over this again when the feature is done.

Reply:
And we cannot just make the InputDictionaryType the base model - because BaseModel only accepts `InputModelType`.